examples/nlp/language_modeling/megatron_lm_pt_to_nemo.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Conversion script to convert Megatron_LM checkpoints into nemo checkpoint.
+  Example to run this conversion script:
+    python -m torch.distributed.launch --nproc_per_node=<tensor_model_parallel_size> megatron_lm_pt_to_nemo.py.py \
+     --checkpoint_folder <path_to_PTL_checkpoints_folder> \
+     --checkpoint_name <checkpoint_name> \
+     --nemo_file_path <path_to_output_nemo_file> \
+     --model_type <megatron model type> \
+     --tensor_model_parallel_size <tensor_model_parallel_size>
+"""
+
+import os
+from argparse import ArgumentParser
+
+import torch
+from pytorch_lightning.trainer.trainer import Trainer
+
+from nemo.collections.nlp.models.language_modeling.megatron_gpt_model import MegatronGPTModel
+from nemo.collections.nlp.models.language_modeling.megatron_bert_model import MegatronBertModel
+
+from nemo.collections.nlp.parts.nlp_overrides import NLPSaveRestoreConnector
+from nemo.utils import AppState, logging
+from typing import Any, Optional
+from pytorch_lightning.utilities.migration import pl_legacy_patch
+from pytorch_lightning.core.saving import load_hparams_from_tags_csv, load_hparams_from_yaml
+from pytorch_lightning.utilities.cloud_io import load as pl_load
+from collections import OrderedDict
+
+
+def get_args():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--checkpoint_folder",
+        type=str,
+        default=None,
+        required=True,
+        help="Path to PTL checkpoints saved during training. Ex: /raid/Megatron_LM/checkpoints",
```

Comment:
I think there's a typo here, it should be "Path to Megatron-LM checkpoints saved ..." ?

Reply:
fixed